Velaptor/Content/Caching/SoundCache.cs

```diff
@@ -0,0 +1,184 @@
+            this.path = path ?? throw new ArgumentNullException(nameof(path), NullParamExceptionMsg);
+
+            if (shutDownReactable is null)
+            {
+                throw new ArgumentNullException(nameof(shutDownReactable), NullParamExceptionMsg);
+            }
+
+            this.shutDownUnsubscriber = shutDownReactable.Subscribe(new Reactor<ShutDownData>(_ => ShutDown()));
+
+            this.disposeSoundsReactable =
+                disposeSoundsReactable ??
+                throw new ArgumentNullException(nameof(disposeSoundsReactable), NullParamExceptionMsg);
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the <see cref="SoundCache"/> class.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        ~SoundCache()
+        {
+            if (UnitTestDetector.IsRunningFromUnitTest)
+            {
+                return;
+            }
+
+            ShutDown();
+        }
+
+        /// <inheritdoc/>
+        public int TotalCachedItems => this.sounds.Count;
+
+        /// <inheritdoc/>
+        public ReadOnlyCollection<string> CacheKeys => new (this.sounds.Keys.ToArray());
+
+        /// <summary>
+        /// Gets a sound using the given <paramref name="soundFilePath"/>.
+        /// </summary>
+        /// <param name="soundFilePath">
+        ///     The full file path to a <c>Sound</c> file.
+        /// </param>
+        /// <returns>A sound loaded from either a <c>'.ogg'</c> or <c>'.mp3'</c> file.</returns>
+        /// <remarks>
+        /// <para>
+        ///     If the item has not been previously created, the <see cref="TextureCache"/> class
+        ///     will retrieve it, and then cache it for fast retrieval.
+        /// </para>
+        /// <para>
+        ///     The only sound files supported are <c>'.ogg'</c> and <c>'.mp3'</c> files.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        ///     Thrown if the <paramref name="soundFilePath"/> is null or empty.
+        /// </exception>
+        /// <exception cref="LoadSoundException">
+        ///     Thrown if the type of sound file being requested is not any of the supported types below:
+        /// <list type="bullet">
+        ///     <item><c>'.ogg'</c></item>
+        ///     <item><c>'.mp3'</c></item>
+        /// </list>
+        /// </exception>
+        /// <exception cref="FileNotFoundException">
+        ///     Thrown if the file at the given <paramref name="soundFilePath"/> is not found.
+        /// </exception>
+        public ISound GetItem(string soundFilePath)
+        {
+            if (string.IsNullOrEmpty(soundFilePath))
+            {
+                throw new ArgumentNullException(nameof(soundFilePath), "The parameter must not be null or empty.");
+            }
+
+            var extension = this.path.GetExtension(soundFilePath);
+
+            if (new[] { OggFileExtension, Mp3FileExtension }.Contains(extension) is false)
```

Comment:
Initiating an array to see if the extension is a valid extension is not performant.

Refactor this code to do basic || comparison operations